polyA/calculate_score.py

```diff
@@ -9,6 +10,7 @@ def calculate_score(
     prev_char_seq1: str,
     prev_char_seq2: str,
     sub_matrix: Dict[str, int],
+    char_complexity_adjustment: Dict[str, int] = None,
```

Comment:
The type here should be `Optional[Dict[str, int]]` since you're setting `None` as the default value.